apps/sms/views/conversation/js/conversation.js

```diff
@@ -721,6 +723,48 @@ var ConversationView = {
     });
   },
 
+  handleComposer: function thui_handleComposer(composer, recipients) {
+
+    function generateMmsAttachmentsForComposer(slides, content) {
+      if (typeof content === 'string') {
+        slides.push({
+          content: new Blob([content], { type: 'text/plain' }),
+        });
+      } else {
+        slides.push({
+          content: content.blob,
+          location: content.name
+        });
+      }
+      return slides;
+    }
+
+    function addRecipientsToConversation(data) {
+      if (data.type !== '') {
+        data.source = 'contacts';
+      }
+      ConversationView.recipients.add(data);
+    }
+
+    Compose.fromMessage({
+      subject: composer.subject,
+      type: composer.type,
+      body: composer.body,
+      attachments: composer.content
+        .reduce(generateMmsAttachmentsForComposer, [])
+    });
+    Compose.focus();
+
+    if (recipients) {
+      var findByAddress = Contacts.findByAddress.bind(Contacts);
+      for (var i in recipients) {
+        Utils.getContactDisplayInfo(findByAddress, recipients[i])
```

Comment:
I don't know if you remember, but you asked me to factorize that with how we get contacts for handleActivity. Actually I think it's not exactly the same, so I kept it like that.

The difference is that in handleActivity when you don't have a activity.contact, it puts a 'manual' source. But here I don't have the contact even if sometimes there may be one in the user DB. So I want to really call findByAddress at some point.

Btw is there a way to get the current MozContact of a conversation? It would allow me to factorize with handleActivity.